File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.java
class Solution {
    public int divisorSubstrings(int num, int k) {
        String s = String.valueOf(num);
        int n = s.length();
        int count = 0;
        for (int i = 0; i + k <= n; i++) {
            String sub = s.substring(i, i + k);
            int val = Integer.parseInt(sub);

            if (val == 0) {
                continue;
            }
            
            if (num % val == 0) {
                count++;
            }
        }
        return count;
    }
}

File: 2269-find-the-k-beauty-of-a-number/2269-find-the-k-beauty-of-a-number.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        
        count = 0

        string_num = str(num)
        n = len(string_num)

        for i in range(n - k + 1):
            j = i + k
            sub = int(string_num[i: j])
            print(sub)
            if sub == 0:
                continue
            if num % sub == 0:
                count += 1
            
        return count
        